rs/node_rewards/rewards_calculation/src/performance_based_algorithm/results.rs

```diff
@@ -5,71 +5,119 @@ use rust_decimal::Decimal;
 pub type Percent = Decimal;
 pub type Region = String;
 
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
 pub struct NodeMetricsDaily {
+    /// The subnet to which this node was assigned on this day
     pub subnet_assigned: SubnetId,
+    /// Subnet Assigned Failure Rate
+    ///
+    /// The failure rate of the entire subnet for this day.
     pub subnet_assigned_fr: Percent,
+    /// Number of blocks proposed by this node on this day
     pub num_blocks_proposed: u64,
+    /// Number of blocks that failed to be included on this day
     pub num_blocks_failed: u64,
+    /// Original Failure Rate
+    ///
+    /// Calculated as `num_blocks_failed / (num_blocks_proposed + num_blocks_failed)`.
+    /// Represents the raw failure rate of the node before any subnet-level adjustments.
     pub original_fr: Percent,
+    /// Relative Failure Rate
+    ///
+    /// Failure rate adjusted for subnet performance.
+    /// Calculated as `max(0, original_fr - subnet_assigned_fr)`.
     pub relative_fr: Percent,
 }
 
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
 pub enum NodeStatus {
+    /// Node is assigned to a subnet with recorded metrics
     Assigned { node_metrics: NodeMetricsDaily },
-    Unassigned { extrapolated_fr: Percent },
+    /// Node is unassigned; only extrapolated failure rate is available
+    Unassigned {
+        /// Extrapolated Failure Rate (EFR)
+        /// Used to estimate the node's performance when unassigned
+        extrapolated_fr: Percent,
+    },
 }
 
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
 pub struct NodeResults {
+    /// Unique identifier of the node
     pub node_id: NodeId,
+    /// NodeRewardType
     pub node_reward_type: NodeRewardType,
+    /// Geographical region of the node
     pub region: String,
+    /// Data center identifier
     pub dc_id: String,
+    /// Node status, assigned or unassigned, with associated metrics
     pub node_status: NodeStatus,
+    /// Performance multiplier (1 - rewards_reduction)
+    ///
+    /// Represents how rewards are adjusted based on node performance
     pub performance_multiplier: Percent,
+    /// Rewards reduction applied due to failure rates
     pub rewards_reduction: Percent,
+    /// Base rewards before applying performance multipliers
     pub base_rewards: XDRPermyriad,
+    /// Rewards adjusted by the performance multiplier
     pub adjusted_rewards: XDRPermyriad,
 }
 
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
 pub struct BaseRewards {
+    /// NodeRewardType
     pub node_reward_type: NodeRewardType,
+    /// Region for which base rewards are calculated
     pub region: Region,
+    /// Monthly base rewards in XDR permyriad
     pub monthly: XDRPermyriad,
+    /// Daily base rewards in XDR permyriad
     pub daily: XDRPermyriad,
 }
 
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
 pub struct BaseRewardsType3 {
```

Comment:
I guess this means that BaseRewards does not cover type 3 nodes? Maybe, name should say so (a pre-existing issue). But at least, comments can say so.